main/src/cgeo/geocaching/CacheListActivity.java

```diff
@@ -1694,15 +1696,17 @@ public static void startActivityFinder(final Activity context, final String user
     }
 
     private void prepareFilterBar() {
-        if (Settings.getCacheType() != CacheType.ALL || adapter.isFiltered()) {
-            final StringBuilder output = new StringBuilder(Settings.getCacheType().getL10n());
-
+        final boolean cacheTypeFilterActive = Settings.getCacheType() != CacheType.ALL;
+        if (cacheTypeFilterActive || adapter.isFiltered()) {
+            final List<String> filters = new ArrayList<>();
+            if (cacheTypeFilterActive) {
```

Comment:
Yes, thought about the ifs, too. Had various versions of this function. I've extracted the creation of the `filterNames` List, so the original method is just responsible for setting the UI state, wdyt?